### PR TITLE
insert the full target triplet in the package name

### LINKED ISSUE #1385: pre-built binaries should include fully qualified target
our pre-built binaries are just called "linux" "macos" and "windows" - it does not call out anything about underlying architectures (`x86` vs. `ARM`) nor about the toolchain (`gnu` vs `musl` vs `msvc`). this information should be included in the tarball release directories.

### PR DESCRIPTION
**Pull request by Geal**
Fix #1385

code extracted from rover's xtask command

I updated the URL in the Dockerfile, but that means it would not work until the next release, should that go into a separate PR instead?

### CODE REVIEW

**Comment by abernix** [comment]
The Docker image itself isn't published until release-time, right?  Is the concern about the time where it lives on `main` and isn't correct? (Is it depended on from GitHub in some way I'm not sure of?). cc @garypen, probably?

File: xtask/src/commands/package/mod.rs [hunk 1]
```diff
@@ -11,6 +13,16 @@ use structopt::StructOpt;
 use xtask::*;
 
 const INCLUDE: &[&str] = &["README.md", "LICENSE", "licenses.html"];
+pub(crate) const TARGET_MUSL_LINUX: &str = "x86_64-unknown-linux-musl";
+pub(crate) const TARGET_GNU_LINUX: &str = "x86_64-unknown-linux-gnu";
+pub(crate) const TARGET_WINDOWS: &str = "x86_64-pc-windows-msvc";
+pub(crate) const TARGET_MACOS: &str = "x86_64-apple-darwin";
```
Comment (garypen) [inline xtask/src/commands/package/mod.rs:19]: Is it worth naming `TARGET_MACOS` `TARGET_X86_64_MACOS` to allow for the eventual arrival of `aarch64-apple-darwin` as `TARGET_ARM_64_MACOS`?

**Review by garypen — APPROVED** [review]
The docker file change is fine. We only do that on release. One naming suggestion which is a NIT and can be ignored if you like.

**Comment by Geal** [comment]
I'm opening #1397 as a follow up, because the diy image construction scripts are likely used from a cloned repository

**Pull request merged**